src/components/Dropdown/Dropdown.tsx

```diff
@@ -259,6 +270,7 @@ export const Dropdown: React.FC<DropdownProps> = ({
                     maxHeight={menuMaxHeight}
                     anchorEl={anchorEl}
                     hasSearch={hasSearch}
+                    autoSearch={autoSearch}
```

Comment:
I was thinking because there's also grouped data that needed to be flattened and I didn't want to handle the logic figuring out what type of data and how to flatten them at this level. 

so I decided to pass that down to the DropdownMenu and TreeViewMenu to do their own flattening logic.

but handling this on the Menu level maybe simplifies things a bit. Will issue a new PR.